src/ai/backend/manager/errors/exceptions.py

```diff
@@ -0,0 +1,1043 @@
+            error_detail=ErrorDetail.INTERNAL_ERROR,
+        )
+
+
+class NotImplementedAPI(BackendError, web.HTTPBadRequest):
+    error_type = "https://api.backend.ai/probs/not-implemented"
+    error_title = "This API is not implemented."
+
+    @classmethod
+    def error_code(cls) -> ErrorCode:
+        return ErrorCode(
+            domain=ErrorDomain.BACKENDAI,
+            operation=ErrorOperation.API,
+            error_detail=ErrorDetail.INTERNAL_ERROR,
+        )
+
+
+class DeprecatedAPI(BackendError, web.HTTPBadRequest):
+    error_type = "https://api.backend.ai/probs/deprecated"
+    error_title = "This API is deprecated."
+
+    @classmethod
+    def error_code(cls) -> ErrorCode:
+        return ErrorCode(
+            domain=ErrorDomain.BACKENDAI,
+            operation=ErrorOperation.API,
+            error_detail=ErrorDetail.BAD_REQUEST,
+        )
+
+
+class InvalidAuthParameters(BackendError, web.HTTPBadRequest):
+    error_type = "https://api.backend.ai/probs/invalid-auth-params"
+    error_title = "Missing or invalid authorization parameters."
+
+    @classmethod
+    def error_code(cls) -> ErrorCode:
+        return ErrorCode(
+            domain=ErrorDomain.USER,
+            operation=ErrorOperation.AUTH,
+            error_detail=ErrorDetail.INVALID_PARAMETERS,
+        )
+
+
+class AuthorizationFailed(BackendError, web.HTTPUnauthorized):
+    error_type = "https://api.backend.ai/probs/auth-failed"
+    error_title = "Credential/signature mismatch."
+
+    @classmethod
+    def error_code(cls) -> ErrorCode:
+        return ErrorCode(
+            domain=ErrorDomain.USER,
+            operation=ErrorOperation.AUTH,
+            error_detail=ErrorDetail.UNAUTHORIZED,
+        )
+
+
+class PasswordExpired(BackendError, web.HTTPUnauthorized):
+    error_type = "https://api.backend.ai/probs/password-expired"
+    error_title = "Password has expired."
+
+    @classmethod
+    def error_code(cls) -> ErrorCode:
+        return ErrorCode(
+            domain=ErrorDomain.USER,
+            operation=ErrorOperation.AUTH,
+            error_detail=ErrorDetail.UNAUTHORIZED,
+        )
+
+
+class InvalidAPIParameters(BackendError, web.HTTPBadRequest):
+    error_type = "https://api.backend.ai/probs/invalid-api-params"
+    error_title = "Missing or invalid API parameters."
+
+    @classmethod
+    def error_code(cls) -> ErrorCode:
+        return ErrorCode(
+            domain=ErrorDomain.BACKENDAI,
+            operation=ErrorOperation.API,
+            error_detail=ErrorDetail.INVALID_PARAMETERS,
+        )
+
+
+class GraphQLError(BackendError, web.HTTPBadRequest):
+    error_type = "https://api.backend.ai/probs/graphql-error"
+    error_title = "GraphQL-generated error."
+
+    @classmethod
+    def error_code(cls) -> ErrorCode:
+        return ErrorCode(
+            domain=ErrorDomain.BACKENDAI,
+            operation=ErrorOperation.API,
+            error_detail=ErrorDetail.INVALID_PARAMETERS,
```

Comment:
Does this error only occurred due to invalid parameter?

Reply:
I've just refactored using agent mode and haven't adjusted the values yet... Please be patient.